$OpenBSD: patch-services_outside_network_c,v 1.1 2012/01/04 21:41:45 sthen Exp $

- Fix for memory leak (about 20 bytes when a tcp or udp send operation
towards authority servers failed, takes about 50.000 such failures to
leak one Mb, such failures are also usually logged), reported by
Robert Fleischmann.

From r2578 upstream.

--- services/outside_network.c.orig	Tue Dec 27 15:24:16 2011
+++ services/outside_network.c	Tue Dec 27 15:24:42 2011
@@ -1824,6 +1824,7 @@ outnet_serviced_query(struct outside_network* outnet,
 			if(!serviced_udp_send(sq, buff)) {
 				(void)rbtree_delete(outnet->serviced, sq);
 				free(sq->qbuf);
+				free(sq->zone);
 				free(sq);
 				free(cb);
 				return NULL;
@@ -1832,6 +1833,7 @@ outnet_serviced_query(struct outside_network* outnet,
 			if(!serviced_tcp_send(sq, buff)) {
 				(void)rbtree_delete(outnet->serviced, sq);
 				free(sq->qbuf);
+				free(sq->zone);
 				free(sq);
 				free(cb);
 				return NULL;
